test/integration/targets/yarn/tasks/run.yml

```diff
@@ -0,0 +1,114 @@
+- name: 'Create directory for Node'
+  file:
+    path: /usr/local/lib/nodejs
+    state: directory
+
+- name: 'Download Nodejs'
+  unarchive:
+    src: 'https://nodejs.org/dist/v{{ nodejs_version }}/{{ nodejs_path }}.tar.gz'
+    dest: '{{ output_dir }}'
+    remote_src: yes
+    creates: '{{ output_dir }}/{{ nodejs_path }}.tar.gz'
+
+- name: 'Download Yarn'
+  unarchive:
+    src: 'https://yarnpkg.com/downloads/{{yarn_version}}/yarn-v{{yarn_version}}.tar.gz'
+    dest: '{{ output_dir }}'
+    remote_src: yes
+    creates: '{{ output_dir }}/yarn-v{{yarn_version}}_pkg.tar.gz'
+
+- name: 'Copy node to directory created earlier'
+  command: "mv {{ output_dir }}/{{ nodejs_path }} /usr/local/lib/nodejs/{{nodejs_path}}"
+
+# Clean up before running tests
+- name: Remove any previous Nodejs modules
+  file:
+    path: '{{output_dir}}/node_modules'
+    state: absent
+
+# Set vars for our test harness
+- vars:
+    #node_bin_path: "/usr/local/lib/nodejs/node-v{{nodejs_version}}/bin"
+    node_bin_path: "/usr/local/lib/nodejs/{{ nodejs_path }}/bin"
+    yarn_bin_path: "{{ output_dir }}/yarn-v{{ yarn_version }}/bin"
+    package: 'iconv-lite'
+  environment:
+    PATH: "{{ node_bin_path }}:{{ansible_env.PATH}}"
+  block:
+
+    # Get the version of Yarn and register to a variable
+    - shell: '{{ yarn_bin_path }}/yarn --version'
+      environment:
+        PATH: '{{ node_bin_path }}:{{ ansible_env.PATH }}'
+      register: yarn_version
+
+    - name: 'Create dummy package.json'
+      template:
```

Comment:
@verkaufer you need copy module, not a template as you don't interpolate any vars there.